src/CMakeLists.txt

```diff
@@ -229,3 +177,19 @@ target_include_directories(parthenon PUBLIC
 )
 
 lint_target(parthenon)
+
+install(TARGETS parthenon LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/parthenon")
+
+# Maintain directory structure in installed include files
+install(FILES ${base_hdr} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+foreach( subdir ${parthenon_subdirs} )
+  file(TO_CMAKE_PATH "${subdir}" hdr_subdir)
+  install(DIRECTORY "${hdr_subdir}" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  FILES_MATCHING PATTERN "*.hpp")
+  unset(hdr_subdir)
+endforeach()
+
+# Install generated config header file
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/generated/config.hpp
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
What happens if `DESTINATION` is not set? Does this external build target machinery turn off? Or does is it undefined behavior?

Reply:
Now that this code is only processed when `CMAKE_INSTALL_PREFIX` is provided I dont think this is a problem.

Reply:
Cool thanks.